doc/pattern_processing.rst

```diff
@@ -67,6 +75,51 @@ Gaussian kernel, ``sigma``:
 
 Patterns are rescaled to fill the available data type range.
 
+.. _pattern-averaging:
+
+Average neighbour patterns
+==========================
+
+With :meth:`~kikuchipy.signals.ebsd.EBSD.average_neighbour_patterns`, the
+signal-to-noise ratio in patterns in an EBSD scan ``s`` can be improved by
+averaging patterns with their closest neighbours within a square kernel:
+
+.. code-block:: python
+
+    >>> s.average_neighbour_patterns(
+            n_neighbours=1, exclude_kernel_corners=True)
+
+The number of nearest neighbours to average with can be set with
+``n_neighbours`` (default is 1), and whether to exclude patterns in corners of
+the averaging kernel can be set with ``exclude_kernel_corners`` (default is
+``True``). Whether a pattern is considered to be in a kernel corner is
+determined by its radial distance to the kernel origin, with the maximum allowed
+distance equal to the kernel half width.
```

Comment:
Perhaps an example of how the kernel looks like with a higher number of neighbours (4+) with and without the exclude corners option would be useful too?

Reply:
Examples of kernels are now included in the user guide.